dialer.go

```diff
@@ -516,12 +552,94 @@ func (d *Dialer) Close() error {
 	close(d.closed)
 	d.lock.Lock()
 	defer d.lock.Unlock()
+	if d.domainNameTicker != nil {
+		d.domainNameTicker.Stop()
+	}
 	for _, i := range d.cache {
 		i.Close()
 	}
 	return nil
 }
 
+func (d *Dialer) pollDomainNames(ctx context.Context) {
+	type cacheEntry struct {
+		cn    instance.ConnName
+		cache monitoredCache
+	}
+
+	// List all the cache entries created with a domain name
+	d.lock.RLock()
+	cacheEntries := make([]cacheEntry, 0, len(d.cache))
+	for cn, cache := range d.cache {
+
+		// Ignore cache entries that were not opened by domain name.
+		if cn.DomainName() == "" {
+			continue
+		}
+
+		cacheEntries = append(cacheEntries, cacheEntry{cn: cn, cache: cache})
+	}
+	d.lock.RUnlock()
+
+	for _, entry := range cacheEntries {
+		// Remove closed sockets from cache.openSockets
+		entry.cache.lock.Lock()
```

Comment:
`mu` is a more typical name in Go and avoids the stutter: lock.Lock()

Reply:
I renamed the mutex fields to `mu`.